### PR TITLE
Documentation for R4 Practitioner Create

### PR DESCRIPTION
**Pull request by FFSol**
PR Checklist
----
- [x] Screenshot(s) of changes attached before changes merged.
- [x] Screenshot(s) of changes attached after changes merged and published.

### CODE REVIEW

**Comment by averyallen** [comment]
We should go ahead and us the PR template that will be merged soon: https://github.com/cerner/fhir.cerner.com/pull/316

File: content/millennium/r4/individuals/practitioner.md [hunk 1]
```diff
@@ -89,6 +89,58 @@ List an individual Practitioner by its id:
...
+
+    POST /Practitioner
+
+### Authorization Types
+
+<%= authorization_types(practitioner: true, patient: false, system: true) %>
+
+### Headers
+
+<%= headers head: {Authorization: '&lt;OAuth2 Bearer Token>', 'Content-Type': 'application/fhir+json'} %>
```
Comment (averyallen) [inline content/millennium/r4/individuals/practitioner.md]: should the gt here be &gt;?

File: lib/resources/example_json/r4_examples_practitioner.rb [hunk 2]
```diff
@@ -166,5 +166,46 @@ module Resources
...
+        {
+          "type": {
+            "coding": [
+              {
+                "code": "DEA",
+                "system": "http://terminology.hl7.org/CodeSystem/v2-0203"
+              }
+            ]
+          },
+          "system": "urn:oid:1.1.1.1.1.1",
```
Comment (averyallen) [inline lib/resources/example_json/r4_examples_practitioner.rb]: We should use the real system that we associate with DEA, urn:oid:2.16.840.1.113883.4.814 

Also, let's have the example match the DEA number format: https://en.wikipedia.org/wiki/DEA_number

File: lib/resources/r4/practitioner.yaml [hunk 3]
```diff
@@ -24,3 +26,117 @@ fields:
...
+    <ul>
+      <li> If provided, each identifier <b>must</b> meet the following conditions:
+        <ul>
+          <li> Must be a unique identifier. </li>
+          <li> Must not contain the `id` or `assigner` fields. </li>
+          <li> If set, `use` must have a value of 'official'. </li>
+          <li> If set, `type` must meet the following conditions:
+            <ul>
+              <li> Must contain exactly one `coding` field. </li>
+              <li> Must be a valid CodeableConcept. </li>
```
Comment (averyallen) [inline lib/resources/r4/practitioner.yaml]: Do we need this line?  I think everything has to be a valid FHIR element.

File: lib/resources/r4/practitioner.yaml [hunk 4]
```diff
@@ -24,3 +26,117 @@ fields:
...
+            </ul>
+          </li>
+          <li> Must contain the `system` and `value` fields. </li>
+          <li> When specifying a `period`, must include a timecomponent with a timezone. </li>
```
Comment (averyallen) [inline lib/resources/r4/practitioner.yaml]: timecomponent -> time component

File: content/millennium/r4/individuals/practitioner.md [hunk 5]
```diff
@@ -89,6 +89,58 @@ List an individual Practitioner by its id:
...
+Last-Modified: Fri, 6 Dec 2019 15:59:30 GMT
+Location: https://fhir-ehr.sandboxcerner.com/r4/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/Practitioner/1994021
+Server-Response-Time: 3890.363996
+Status: 201 Created
+Vary: Origin
+X-Request-Id: 1638e30e497b93ff4383b2ff0eaeea68
+X-Runtime: 3.890282
+</pre>
+
+The `ETag` response header indicates the current `If-Match` version to use on subsequent updates.
```
Comment (averyallen) [inline content/millennium/r4/individuals/practitioner.md]: Do we normally document this if we don't support updates yet?

**Comment by FFSol** [comment]
Template applied, screenshots incoming with PR comment resolution

File: content/millennium/r4/individuals/practitioner.md (hunk 1 above)
Comment (FFSol) [reply content/millennium/r4/individuals/practitioner.md]: https://github.com/cerner/fhir.cerner.com/pull/330/commits/c128b759084993b859b499ef4e9e83aedf8a4898#diff-00a306a2aa1334ff6f799b4200c75030R104

File: content/millennium/r4/individuals/practitioner.md (hunk 5 above)
Comment (FFSol) [reply content/millennium/r4/individuals/practitioner.md]: Great catch, removed:
https://github.com/cerner/fhir.cerner.com/pull/330/commits/c128b759084993b859b499ef4e9e83aedf8a4898#diff-00a306a2aa1334ff6f799b4200c75030L138

File: lib/resources/example_json/r4_examples_practitioner.rb (hunk 2 above)
Comment (FFSol) [reply lib/resources/example_json/r4_examples_practitioner.rb]: Used correct system and applied an actual DEA number. 

https://github.com/cerner/fhir.cerner.com/pull/330/commits/c128b759084993b859b499ef4e9e83aedf8a4898#diff-b7b6f9bc4e73a723f2d414890a03bde8R200

The current format of the DEA Number linked actually has an incorrect calculation for the checksum on the last number (had it listed as 1234567, when it should be 1234563), so I edited the wikipedia entry too 😅

File: lib/resources/r4/practitioner.yaml (hunk 3 above)
Comment (FFSol) [reply lib/resources/r4/practitioner.yaml]: Trimmed and consolidated
https://github.com/cerner/fhir.cerner.com/pull/330/commits/c128b759084993b859b499ef4e9e83aedf8a4898#diff-c593d32e3c230e494be00194f16e6986L118

File: lib/resources/r4/practitioner.yaml (hunk 4 above)
Comment (FFSol) [reply lib/resources/r4/practitioner.yaml]: Corrected misspelling in [Practitioner](https://github.com/cerner/fhir.cerner.com/pull/330/commits/c128b759084993b859b499ef4e9e83aedf8a4898#diff-c593d32e3c230e494be00194f16e6986R120) and [Patient](https://github.com/cerner/fhir.cerner.com/pull/330/commits/c128b759084993b859b499ef4e9e83aedf8a4898#diff-07c686ec4248977e084ba5a602c761ceR165).

File: content/millennium/r4/individuals/practitioner.md (hunk 1 above)
Comment (MaxPhilips) [reply content/millennium/r4/individuals/practitioner.md]: to my knowledge it does not have to be

File: lib/resources/r4/practitioner.yaml [hunk 6]
```diff
@@ -24,3 +26,112 @@ fields:
...
+          }
+        }
+      ]
+    }
+  note: |
+    <ul>
+      <li> Exactly one name <b>must</b> be provided. The provided name must meet the following conditions:
+        <ul>
+          <li> Must include both `given` and `family`. </li>
+          <li> Up to two `given` names may be provided. If additional given names are required they must be provided in the second value as a single string separated by blank spaces. </li>
```
Comment (MaxPhilips) [inline lib/resources/r4/practitioner.yaml]: maybe 'Up to two values may be present in the `given` names array. If more than two given names are required the second value in the array accepts whitespace-delimited strings.' ?

does the number of whitespaces matter?

File: lib/resources/r4/practitioner.yaml [hunk 7]
```diff
@@ -24,3 +26,112 @@ fields:
...
+  note: |
+    <ul>
+      <li> If provided, each identifier <b>must</b> meet the following conditions:
+        <ul>
+          <li> Must be a unique identifier. </li>
+          <li> Must not contain the `id` or `assigner` fields. </li>
+          <li> If set, `use` must have a value of 'official'. </li>
+          <li> If set, `type` must contain exactly one `coding` field. </li>
+          <li> Must contain the `system` and `value` fields. </li>
+          <li> When specifying a `period`, must include a time component with a timezone. </li>
```
Comment (MaxPhilips) [inline lib/resources/r4/practitioner.yaml]: add 'the fields' as above

File: lib/resources/r4/practitioner.yaml (hunk 6 above)
Comment (FFSol) [reply lib/resources/r4/practitioner.yaml]: > does the number of whitespaces matter?

Each value in that array is limited to 100 characters, so number of whitespaces matters from a functional perspective **there**. A Practitioner could be written with a second given name like `John <insert 80 spaces> Ryan` and we wouldn't fail or anything

File: lib/resources/r4/practitioner.yaml [hunk 8]
```diff
@@ -6,7 +6,8 @@ fields:
   required: 'No'
   cardinality: 0..*
   type: Identifier
-  description: The identifier for the practitioner.
+  description: The identifier(s) for the practitioner.
+  action: terminology
```
[the remark this answers is not in the document]
Comment (FFSol) [reply lib/resources/r4/practitioner.yaml]: Necessary for differentiating between `terminology` identifier and `create` identifier

File: content/millennium/r4/individuals/practitioner.md [hunk 9]
```diff
@@ -89,6 +89,56 @@ List an individual Practitioner by its id:
...
+
+### Example
+
+#### Request
+
+    POST https://fhir-ehr.sandboxcerner.com/r4/0b8a0111-e8e6-4c26-a91c-5069cbc6b1ca/Practitioner
+
+#### Body
+
+  <%= json(:r4_practitioner_create) %>
```
[the remark this answers is not in the document]
Comment (FFSol) [reply content/millennium/r4/individuals/practitioner.md]: https://github.com/cerner/fhir.cerner.com/pull/330/commits/7a2bf1e0a6700f184afdeff837657a6d58827de5#diff-00a306a2aa1334ff6f799b4200c75030R118

File: lib/resources/r4/practitioner.yaml [hunk 10]
```diff
@@ -24,3 +26,112 @@ fields:
...
+          ],
+          "period": {
+            "start": "2019-12-01T00:00:00.000Z"
+          }
+        }
+      ]
+    }
+  note: |
+    <ul>
+      <li> Exactly one name <b>must</b> be provided. The provided name must meet the following conditions:
```
[the remark this answers is not in the document]
Comment (FFSol) [reply lib/resources/r4/practitioner.yaml]: https://github.com/cerner/fhir.cerner.com/pull/330/commits/7a2bf1e0a6700f184afdeff837657a6d58827de5#diff-00a306a2aa1334ff6f799b4200c75030R118

File: lib/resources/r4/practitioner.yaml (hunk 6 above)
Comment (FFSol) [reply lib/resources/r4/practitioner.yaml]: https://github.com/cerner/fhir.cerner.com/pull/330/commits/7a2bf1e0a6700f184afdeff837657a6d58827de5#diff-c593d32e3c230e494be00194f16e6986R74

File: lib/resources/r4/practitioner.yaml (hunk 7 above)
Comment (FFSol) [reply lib/resources/r4/practitioner.yaml]: https://github.com/cerner/fhir.cerner.com/pull/330/commits/7a2bf1e0a6700f184afdeff837657a6d58827de5#diff-c593d32e3c230e494be00194f16e6986R120

File: lib/resources/r4/practitioner.yaml (hunk 8 above)
[the remark this answers is not in the document]
Comment (MaxPhilips) [reply lib/resources/r4/practitioner.yaml]: if you leave it out, won't it render in both tables?

[the remark this answers is not in the document]
Comment (FFSol) [reply lib/resources/r4/practitioner.yaml]: https://github.com/cerner/fhir.cerner.com/pull/330/commits/8ce062c87f65ccf958e50168a9b1afcdeeaa862e

**Comment by FFSol** [comment]
@averyallen @MaxPhilips screenshots added

**Comment by MaxPhilips** [comment]
moving pre-merge screenshots to comment

Screen Shot 2019-12-10 at 10 12 24 AM
----
Screen Shot 2019-12-10 at 10 13 02 AM
----
Screen Shot 2019-12-10 at 10 14 15 AM
----
Screen Shot 2019-12-10 at 10 14 28 AM
----
Screen Shot 2019-12-10 at 10 14 47 AM
----
Screen Shot 2019-12-10 at 10 14 56 AM
----
Screen Shot 2019-12-10 at 10 15 09 AM

**Comment by MF068815** [comment]
**fhir.cerner.com Validated**

Create section is present and appears to display as expected
Screen Shot 2019-12-11 at 10 13 26 AM
Screen Shot 2019-12-11 at 10 13 32 AM

**Pull request merged**